src/v/kafka/server/connection_context.h

```diff
@@ -81,6 +82,7 @@ class request_tracker {
     net::server_probe& _probe;
     handler_probe& _h_probe;
     bool _errored{false};
+    std::unique_ptr<hdr_hist::measurement> _handler_latency;
```

Comment:
should we reduce mem footprint of the histogram to be at millisecond level vs microsecond level for these.

Reply:
posting from slack thread. ```
Alexander Gallego (PDT; San Francisco)
:spiral_calendar_pad:  [5 minutes ago](https://redpandadata.slack.com/archives/C04B8659KGR/p1689104542392729?thread_ts=1689103791.967429&cid=C04B8659KGR)
wthat's the memory footprint hit for these? 185kib


Alexander Gallego (PDT; San Francisco)
:spiral_calendar_pad:  [3 minutes ago](https://redpandadata.slack.com/archives/C04B8659KGR/p1689104616067999?thread_ts=1689103791.967429&cid=C04B8659KGR)
[@Brandon Allard](https://redpandadata.slack.com/team/U03EUPRR5EH)
 should we have typedefs one for high_latency_histogram  or maybe ms_latency_histogram and then one fore second_latency_histogram and one for low latency, us_latency_histogram


Alexander Gallego (PDT; San Francisco)
:spiral_calendar_pad:  [3 minutes ago](https://redpandadata.slack.com/archives/C04B8659KGR/p1689104631525999?thread_ts=1689103791.967429&cid=C04B8659KGR)
so we use smaller memory buckets in the ctor of the histogram?


Alexander Gallego (PDT; San Francisco)
:spiral_calendar_pad:  [3 minutes ago](https://redpandadata.slack.com/archives/C04B8659KGR/p1689104649796919?thread_ts=1689103791.967429&cid=C04B8659KGR)
i imagine the metrics endpoints will always be in the ms and we wouldn't care if it is in the microsecond scale```

Reply:
One concern I have about reducing the range of the histogram from 1us-1hr to 1ms-1hr is that for some kafka methods that don't require cross-node communication like metadata requests we'd expect sub-ms latencies under normal load.

However, even then the only time we'd have concerns about metadata requests and care about the latency is when it is exhibiting 1ms+ latencies. I.e, when we're trying to return the metadata for 1k+ partitions frequently.

So as long as we don't mind the lower bound latency of all our handler histograms showing as 1ms even though they are probably lower than that I suppose the 1ms-1hr range would still give us the signal we are looking for when it comes to debugging perf issues.